lib/streamlit/config.py

```diff
@@ -1015,7 +959,7 @@ def _browser_server_port() -> int:
     scriptable=True,
     deprecated=True,
     deprecation_text="The support for global pyplot instances is planned to be removed soon.",
-    expiration_date="2024-04-15",
+    expiration_date="2024-12-15",
```

Comment:
I'm not a fan of moving this date to the end of the year. Especially since we keep the actual deprecation date of the pyplot global usage option at `December 1st, 2020`:

https://github.com/streamlit/streamlit/blob/a7ac4decd79336e83e7721a2fcd1a582a2051d31/lib/streamlit/elements/pyplot.py#L194

It's about time to decide whether to keep the feature or remove it. The global object for pyplot is used in 3.3% of `st.pyplot` command usage (-> 0.1% of all apps). For whatever reason, this is a lot lower than the usage of the `deprecation.showPyplotGlobalUse` config option, which is at 0.6%. I did check on Github for `deprecation.showPyplotGlobalUse,`, and it seems that there are quite a few apps that unnecessarily use this config but are actually not using the deprecated feature. 

My suggestion: Let's decide whether we want to keep the feature or follow through on removing it. If we want to keep it, let's remove the deprecation warning altogether. If not, let's remove the `deprecation.showPyplotGlobalUse` config option and update the code here to use our deprecation utility function so that its only shown in development mode:

https://github.com/streamlit/streamlit/blob/a7ac4decd79336e83e7721a2fcd1a582a2051d31/lib/streamlit/elements/pyplot.py#L108-L109

Reply:
But I think this doesn't need to be part of this PR; maybe we keep this one change out of this PR and make a change based on the decision in a dedicated PR.

Reply:
I discussed it with @jrieke, and he agrees to remove the config option and change it to use our deprecation utils. I will do it in a separate PR.